### PR TITLE
[magma] add port

### PR DESCRIPTION
**Pull request by Neumann-A**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html)
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [no usage] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by Neumann-A** [comment]
Hmm CUDA detection fails in linux CI?

**Comment by MonicaLiu0311** [comment]
* Do you need to put the header files in the `include/magma/` folder to prevent future conflicts?
   image


* Header file not found when testing usage:
   ```
   >------ Build All started: Project: CMakeUsage, Configuration: x64-Debug ------
     MSBuild version 17.7.2+d6990bcfa for .NET Framework
     
       Checking Build System
       CMakeUsage.cpp
   C:\Users\monica\source\repos\CMakeUsage\CMakeUsage\CMakeUsage.cpp(2,10): fatal error C1083: Cannot open include file: 'magma.h': No such file or directory 
   
   Build All failed.
   ```
   <details>
   <summary><b>CMakeUsage.cpp</b></summary>
   <pre>
   #include &#60;iostream&#62;
   #include &#60;magma.h&#62;
   
   using namespace std;
   
   int main()
   {
           cout << "Hello CMake." << endl;
           return 0;
   }
   </pre></details>
   
   
   <details>
   <summary><b>CMakeLists.txt</b></summary>
   <pre>
   cmake_minimum_required (VERSION 3.8)
   
   set(CMAKE_TOOLCHAIN_FILE "G:/magma/scripts/buildsystems/vcpkg.cmake")
   set(PKG_CONFIG_EXECUTABLE "${VCPKG_INSTALLED_DIR}/x64-windows/tools/pkgconf/pkgconf.exe")
   
   project ("CMakeUsage")
   
   add_library (main "CMakeUsage.cpp")
   
   find_package(PkgConfig REQUIRED) 
   pkg_check_modules(MAGMA REQUIRED IMPORTED_TARGET magma) 
   target_include_directories(main PRIVATE ${MAGMA_INCLUDE_DIRS}) 
   target_link_libraries(main ${MAGMA_LIBRARIES})
   </pre></details>

**Comment by Neumann-A** [comment]
> Do you need to put the header files in the include/magma/ folder to prevent future conflicts?

Why? Every magma header is prefixed with `magma_`

> Header file not found when testing usage:

Yeah I only made sure the pc file has `includedir` but i didn't make sure it actually has CFlags.

**Comment by Neumann-A** [comment]
Fixed the include but your usage needs to include `magma_v2.h` and you might need to also add the CUDA include dirs. With the current CUDA approach in vcpkg I simply cannot add the correct include dir for it since it would point to an absolute path outside of vcpkg.

**Comment by MonicaLiu0311** [comment]
> Fixed the include but your usage needs to include `magma_v2.h` and you might need to also add the CUDA include dirs. With the current CUDA approach in vcpkg I simply cannot add the correct include dir for it since it would point to an absolute path outside of vcpkg.

After specifying the CUDA header file path in the system, the PC file test passes:
```
   find_package(PkgConfig REQUIRED) 
   pkg_check_modules(MAGMA REQUIRED IMPORTED_TARGET magma) 
   target_include_directories(main PRIVATE ${MAGMA_INCLUDE_DIRS}) 
   target_link_libraries(main ${MAGMA_LIBRARIES})
```

   <details>
   <summary><b>CMakeUsage.cpp</b></summary>
   <pre>
   
   #include &#60;iostream&#62;
   #include &#60;magma.h&#62;
   #include &#60;magma_v2.h&#62;
   #include &#60;cuda_runtime.h&#62;
   
   using namespace std;
   
   int main()
   {
           cout << "Hello CMake." << endl;
           return 0;
   }
   </pre></details>
   
   
   <details>
   <summary><b>CMakeLists.txt</b></summary>
   <pre>
   
   cmake_minimum_required (VERSION 3.8)
   
   set(CMAKE_TOOLCHAIN_FILE "G:/magma/scripts/buildsystems/vcpkg.cmake")
   set(PKG_CONFIG_EXECUTABLE "${VCPKG_INSTALLED_DIR}/x64-windows/tools/pkgconf/pkgconf.exe")
   include_directories("C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v12.2/include")
   
   project ("CMakeUsage")
   
   add_library (main "CMakeUsage.cpp")
   
   find_package(PkgConfig REQUIRED) 
   pkg_check_modules(MAGMA REQUIRED IMPORTED_TARGET magma) 
   target_include_directories(main PRIVATE ${MAGMA_INCLUDE_DIRS}) 
   target_link_libraries(main ${MAGMA_LIBRARIES})
   </pre></details>

**Review by MonicaLiu0311 — APPROVED** [review]

File: scripts/ci.baseline.txt [hunk 1]
```diff
@@ -678,6 +678,7 @@ log4cpp:x64-linux=fail # dynamic exception specifications
 loguru:arm-neon-android=fail
 loguru:arm64-android=fail
 loguru:x64-android=fail
+magma:x64-linux=fail
```
Comment (BillyONeal) [inline scripts/ci.baseline.txt:681]: Is this due to the CUDA thing?

Comment (Neumann-A) [reply scripts/ci.baseline.txt:681]: Yes. For some reason CMake is unable to find CUDA -> see https://dev.azure.com/vcpkg/public/_build/results?buildId=98131&view=logs&jobId=f79cfdd7-47a8-597f-8f57-dc3e21a8f2ad

**Pull request merged**